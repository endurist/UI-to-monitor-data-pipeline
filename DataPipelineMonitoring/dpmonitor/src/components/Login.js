import React, { useState } from 'react';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import Navbar from 'react-bootstrap/Navbar';
import PropTypes from 'prop-types';
import './Login.css';

export default function Login({ setToken }) {
    const [unityid, setUnityid] = useState();

    // No authentication, LAS will implement their system later
    // Just set the token to the entered unity id
    const handleSubmit = (e) => {
        e.preventDefault();
        const token = { unityid };
        setToken(token);
    };

    return (
        <div>
            <header>
                <Navbar
                    collapseOnSelect
                    expand='lg'
                    bg='light'
                    className='header-container'
                >
                    <Navbar.Brand className='brand-container'>
                        <a
                            href='//www.ncsu.edu/'
                            rel='noreferrer'
                            target='_blank'
                        >
                            <img
                                src='https://ncsu-las.org/wp-content/themes/awi/img/logo-ncsu.png'
                                srcSet='https://ncsu-las.org/wp-content/themes/awi/img/logo-ncsu.png 1x, https://ncsu-las.org/wp-content/themes/awi/img/logo-ncsu@2x.png 2x'
                                target='_blank'
                                className='logo'
                                alt='NC State University'
                                title='NC State University'
                                width='105'
                                height='50'
                            />
                        </a>
                        <a
                            href='https://ncsu-las.org'
                            rel='noreferrer'
                            target='_blank'
                        >
                            <img
                                src='https://ncsu-las.org/wp-content/themes/awi/img/logo-las.png'
                                srcSet='https://ncsu-las.org/wp-content/themes/awi/img/logo-las.png 1x, https://ncsu-las.org/wp-content/themes/awi/img/logo-las@2x.png 2x'
                                target='_blank'
                                className='logo'
                                alt='Laboratory for Analytic Sciences'
                                title='Laboratory for Analytic Sciences'
                                width='215'
                                height='50'
                            />
                        </a>
                    </Navbar.Brand>
                </Navbar>
            </header>
            <div className='Login'>
                <Form onSubmit={handleSubmit}>
                    <Form.Group
                        size='lg'
                        controlId='unityid'
                        placeholder='Enter unity id'
                    >
                        <Form.Label>UnityID</Form.Label>
                        <Form.Control
                            autoFocus
                            onChange={(e) => setUnityid(e.target.value)}
                        />
                    </Form.Group>

                    <div className='form-group'>
                        <div className='custom-control custom-checkbox'>
                            <input
                                type='checkbox'
                                className='custom-control-input'
                                id='customCheck1'
                            />
                            <label
                                className='custom-control-label'
                                htmlFor='customCheck1'
                            >
                                Remember me
                            </label>
                        </div>
                    </div>
                    <Button block size='lg' type='submit'>
                        Login
                    </Button>
                </Form>
            </div>
        </div>
    );
}

Login.propTypes = {
    setToken: PropTypes.func.isRequired,
};
