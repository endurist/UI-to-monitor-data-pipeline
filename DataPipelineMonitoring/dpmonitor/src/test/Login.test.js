import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { shallow } from 'enzyme';

import Login from '../components/Login';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';

afterEach(cleanup);

const mockSetToken = jest.fn();

it('renders', () => {
    shallow(<Login setToken={mockSetToken} />);
});

it('should take a snapshot', () => {
    const { asFragment } = render(<Login setToken={mockSetToken} />);

    expect(asFragment(<Login setToken={mockSetToken} />)).toMatchSnapshot();
});

it('should render one Form', () => {
    const wrapper = shallow(<Login setToken={mockSetToken} />);
    expect(wrapper.find(Form)).toHaveLength(1);
});

it('should render one Form.Label', () => {
    const wrapper = shallow(<Login setToken={mockSetToken} />);
    expect(wrapper.find(Form.Label)).toHaveLength(1);
});

it('should render one Form.Control', () => {
    const wrapper = shallow(<Login setToken={mockSetToken} />);
    expect(wrapper.find(Form.Control)).toHaveLength(1);
});

it('should render one Button', () => {
    const wrapper = shallow(<Login setToken={mockSetToken} />);
    expect(wrapper.find(Button)).toHaveLength(1);
});

it('should say UnityID on the page', () => {
    const { getByText } = render(<Login setToken={mockSetToken} />);
    expect(getByText('UnityID')).toBeInTheDocument();
});
