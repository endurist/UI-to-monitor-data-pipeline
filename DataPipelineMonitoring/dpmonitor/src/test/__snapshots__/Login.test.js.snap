// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <header>
      <div
        class="header-container"
      >
        <a
          href="//www.ncsu.edu/"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="NC State University"
            class="logo"
            height="50"
            src="https://ncsu-las.org/wp-content/themes/awi/img/logo-ncsu.png"
            srcset="https://ncsu-las.org/wp-content/themes/awi/img/logo-ncsu.png 1x, https://ncsu-las.org/wp-content/themes/awi/img/logo-ncsu@2x.png 2x"
            target="_blank"
            title="NC State University"
            width="105"
          />
        </a>
        <a
          href="https://ncsu-las.org"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Laboratory for Analytic Sciences"
            class="logo"
            height="50"
            src="https://ncsu-las.org/wp-content/themes/awi/img/logo-las.png"
            srcset="https://ncsu-las.org/wp-content/themes/awi/img/logo-las.png 1x, https://ncsu-las.org/wp-content/themes/awi/img/logo-las@2x.png 2x"
            target="_blank"
            title="Laboratory for Analytic Sciences"
            width="215"
          />
        </a>
      </div>
    </header>
    <div
      class="Login"
    >
      <form
        class=""
      >
        <div
          class="form-group"
          placeholder="Enter unity id"
        >
          <label
            class="form-label"
            for="unityid"
          >
            UnityID
          </label>
          <input
            class="form-control"
            id="unityid"
          />
        </div>
        <div
          class="form-group"
        >
          <div
            class="custom-control custom-checkbox"
          >
            <input
              class="custom-control-input"
              id="customCheck1"
              type="checkbox"
            />
            <label
              class="custom-control-label"
              for="customCheck1"
            >
              Remember me
            </label>
          </div>
        </div>
        <button
          class="btn btn-primary btn-block btn-lg"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
